data/src/gadm/preprocess_gadm.py

```diff
@@ -7,150 +7,193 @@
+    df = pd.read_excel(xlsx_path, sheet_name="Countries")
+
+    country_dataset = prepare_adm_datasets(adm0_gdf, adm1_gdf)
+
+    final = gpd.GeoDataFrame(
+        (
+            country_dataset.pipe(prepare_geometries)
+            .pipe(filter_empty_geometries)
+            .pipe(compute_area)
+            .merge(df, on="country_code", how="right")
+        ),
+        crs="epsg:4326",
+    )
     # Save the cleaned  and simplified data
-    adm0_gdf.to_file("../../raw_data/gadm/gadm_410-levels_ADM_0_cleaned.geojson", driver="GeoJSON")
-    print("Cleaned data saved at: ../../raw_data/gadm/gadm_410-levels_ADM_0_cleaned.geojson")
+    print(
+        "Saving the cleaned data to a csv in: ",
+        gpkg_path.parent.joinpath("countries.csv").as_posix(),
+    )
+    final.dropna(subset="geometry").to_wkb(hex=True).to_csv(
+        gpkg_path.parent.joinpath("countries.csv"),
+        index=False,
+    )
 
-    return adm0_gdf
+    return final
 
 
-def fix_geometry(geom):
-    # Validate and fix geometry
-    if not geom.is_valid:
-        geom = make_valid(geom)
-    # Ensure the geometry is in MultiPolygon format
-    if geom.geom_type == "Polygon":
-        geom = MultiPolygon([geom])
-    return geom
+def prepare_geometries(gdf: gpd.GeoDataFrame) -> gpd.GeoDataFrame:
+    """
+    Fix geometries for countries with invalid geometries.
+    """
+    gdf.geometry = (
+        gdf.geometry.make_valid()
+        .simplify(0.01, preserve_topology=True)
+        .buffer(0)
+        .make_valid()
+        .apply(lambda geom: MultiPolygon([geom]) if geom.geom_type == "Polygon" else geom)
+    )
+    return gdf
 
 
-def simplify_gdf(gdf, simplification_tolerance=0.01, preserve_topology=True):
+def filter_empty_geometries(gdf: gpd.GeoDataFrame) -> gpd.GeoDataFrame:
     """
-    Simplifies the geometries of a GeoDataFrame in place and returns it.
+    Filter out empty geometries.
+    """
+    return gdf[~gdf.geometry.is_empty].dropna(subset=["geometry"])
 
-    Parameters:
-    - gdf (GeoDataFrame): The original GeoDataFrame with geometries to simplify.
-    - simplification_tolerance (float): Tolerance for simplification (e.g., 0.01 for 1% simpl).
-    - preserve_topology (bool): If True, attempts to preserve the topology.
 
-    Returns:
-    - GeoDataFrame: A new GeoDataFrame with simplified geometries.
+def compute_area(
+    gdf: gpd.GeoDataFrame, area_col: str = "area_ha", code_col: str = "country_code"
+) -> gpd.GeoDataFrame:
+    """
+    Compute area of geometries. using EPSG:6933 that is an equal area projection. for area calculation.
     """
-    # Simplify the geometries
-    simplified_gdf = gdf.copy()
-    simplified_gdf["geometry"] = gdf["geometry"].simplify(
-        simplification_tolerance, preserve_topology=preserve_topology
+    # we ensure that the areas of this countries match the area used in the excel sheet
+    country_size_ha = {
+        "USA": 947084624.2706754,
```

Comment:
This seems evil. Why are we hardcoding areas for specific countries? 

I guess these countries are the ones that we can **_now_** calculate stuff for.

What happens whenever the client gets more data for additional countries? We will be using the values computed below. It does not seem right